### PR TITLE
Fix stats rendering

### PR DESCRIPTION
**Pull request by trevordmiller**
# Changes

- Fix last X months revenue count to work with Jason's PR (https://github.com/eggheadio/egghead-instructor-center/pull/114); now shows "Last 5 months"; this is dynamic for the amount of months returned by `revenue_url` minus current
- Remove padding on prompt and apply to instances
- Add prompt fallback when no revenue exists
- Don't try to call `instructor.revenue_url` unless it exists

# Result For User

## When there are views/revenue to show

screen shot 2017-02-27 at 2 28 56 pm

## When there are no views/revenue to show

screen shot 2017-02-27 at 2 24 50 pm

## When you don't have access to stats
screen shot 2017-02-27 at 2 24 31 pm

---

### CODE REVIEW

File: src/screens/Overview/components/InstructorLessons/index.js [hunk 1]
```diff
@@ -9,7 +9,7 @@ import {
 } from 'utils/text'
 import {inProgressLessonStates, publishedLessonStates} from 'utils/lessonStates'
 import LessonList from 'components/LessonList'
-import Prompt from './components/Prompt'
+import Prompt from 'components/Prompt'
```
Comment (trevordmiller) [inline src/screens/Overview/components/InstructorLessons/index.js:12]: Promoting Prompt

File: src/screens/Overview/components/InstructorLessons/index.js [hunk 2]
```diff
@@ -26,11 +26,13 @@ export default ({instructor}) => (
...
-                actionText={newLessonsActionText}
-                action={'/lessons/new'}
-              />
+              <div className='pa3'>
+                <Prompt
+                  description={noInProgressLessonsDescriptionText}
+                  actionText={newLessonsActionText}
+                  action={'/lessons/new'}
+                />
+              </div>
```
Comment (trevordmiller) [inline src/screens/Overview/components/InstructorLessons/index.js:35]: Remove side effect padding from Prompt and use in instances instead

File: src/screens/Overview/components/Stats/utils/removeRevenueMonth/index.test.js [hunk 3]
```diff
@@ -1,4 +1,4 @@
-import removeCurrentMonth from '.'
+import removeRevenueMonth from '.'
```
Comment (trevordmiller) [inline src/screens/Overview/components/Stats/utils/removeRevenueMonth/index.test.js:1]: Rename this as it doesn't have to do with the current month, but rather removing a month from the revenue object

File: src/screens/Overview/components/Stats/index.js [hunk 4]
```diff
@@ -46,15 +54,20 @@ export default ({instructor, revenue}) => {
...
-              title={`Last ${size(revenue)} months`}
-              revenue={currentTotalRevenue.revenue}
-              subscriberMinutes={currentTotalRevenue.minutes_watched}
-            />
-          </div>
-        : <div className='mt3 green'>
-            {noRevenueDescriptionText}
+            {currentTotalRevenue && currentTotalRevenue.revenue > 0
+              ? <RevenuePeriod
+                  title={`Last ${currentTotalRevenue.monthCount} months`}
```
Comment (tayiorbeii) [inline src/screens/Overview/components/Stats/index.js:59]: conditionally plural "month"/"months" ?

File: src/utils/text.js [hunk 5]
```diff
@@ -51,7 +51,8 @@ export const revenueLabelText = 'dollars'
 
 export const subscriberMinutesLabelText = 'minutes watched'
 
-export const noRevenueDescriptionText = `There's no revenue to report. Publishing lessons is the key.`
+export const noRevenueDescriptionText = `There's no stats to report. Publishing lessons is the key.`
```
Comment (tayiorbeii) [inline src/utils/text.js:54]: this would technically be "There are" 🤓  so maybe something like "No stats to report! Publishing lessons is the key."

**Review by tayiorbeii — APPROVED** [review]

**Review by zacjones93 — APPROVED** [review]

File: src/screens/Overview/components/Stats/index.js (hunk 4 above)
Comment (trevordmiller) [reply src/screens/Overview/components/Stats/index.js:59]: Good idea. Will be resolved when we upgrade to a proper localization library. Until then shouldn't be an issue as we will show more than 1 month.

File: src/utils/text.js (hunk 5 above)
Comment (trevordmiller) [reply src/utils/text.js:54]: Fixed in next PR.

**Pull request merged**